### ISSUE TITLE
I thinkYour code is taking up unnecessary 6GB memory in selftraining.py

### ISSUE THREAD

**Issue by okjer** [OasisYang/SSG#20]
you don't need cluster_list on line 366,it takes up 6GB,beacuse cluster.components_ is a is a matrix shape of (15xxx,16552),you can see this in site-packages/sklearn/cluster/dbscan_.py.
I think eps_list can have the same effect.
I don't know if I am right, please tell me

**Comment by PCJ936968** [comment]
> you don't need cluster_list on line 366,it takes up 6GB,beacuse cluster.components_ is a is a matrix shape of (15xxx,16552),you can see this in site-packages/sklearn/cluster/dbscan_.py.
> I think eps_list can have the same effect.
> I don't know if I am right, please tell me

hello，Have you verified your conjecture?

**Comment by okjer** [comment]
yes,I try my code and get the same result,and it's evident right




------------------&nbsp;原始邮件&nbsp;------------------
发件人: "One J"<notifications@github.com&gt;; 
发送时间: 2020年3月28日(星期六) 晚上10:39
收件人: "OasisYang/SSG"<SSG@noreply.github.com&gt;; 
抄送: "朱方琪"<398384796@qq.com&gt;; "Author"<author@noreply.github.com&gt;; 
主题: Re: [OasisYang/SSG] I thinkYour code is taking up unnecessary 6GB memory in selftraining.py (#20)





  
you don't need cluster_list on line 366,it takes up 6GB,beacuse cluster.components_ is a is a matrix shape of (15xxx,16552),you can see this in site-packages/sklearn/cluster/dbscan_.py.
 I think eps_list can have the same effect.
 I don't know if I am right, please tell me
  
hello，Have you verified your conjecture?

**Comment by doris1996** [comment]
> 
> yes,I try my code and get the same result,and it's evident right
> […](#)
> ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "One J"<notifications@github.com&gt;; 发送时间: 2020年3月28日(星期六) 晚上10:39 收件人: "OasisYang/SSG"<SSG@noreply.github.com&gt;; 抄送: "朱方琪"<398384796@qq.com&gt;; "Author"<author@noreply.github.com&gt;; 主题: Re: [OasisYang/SSG] I thinkYour code is taking up unnecessary 6GB memory in selftraining.py (#20) you don't need cluster_list on line 366,it takes up 6GB,beacuse cluster.components_ is a is a matrix shape of (15xxx,16552),you can see this in site-packages/sklearn/cluster/dbscan_.py. I think eps_list can have the same effect. I don't know if I am right, please tell me hello，Have you verified your conjecture? — You are receiving this because you authored the thread. Reply to this email directly, view it on GitHub, or unsubscribe.

hi, I'm new to this. So I'm wondering what eps_list is and what you did in detail. By the way, I can't find cluster_list used on line 366 in selftraining.py, did I do something wrong? 
Looking forward to your reply. May help a lot to me. @okjer

**Comment by okjer** [comment]
作者改了代码，你debug一下代码就知道怎么做了

**Comment by doris1996** [comment]
> 
> 
> 作者改了代码，你debug一下代码就知道怎么做了

感谢您的回答，但我确实是个小白，所以现在还是一头雾水
